### DISCUSSION TITLE
CSS nesting issue

### DISCUSSION THREAD

**Discussion by kai-wang** [tailwindlabs/tailwindcss#8695] [Help]
Hi there,

can someone please advise what the potential issue is? 

I have truncated my source file main.css as below

@tailwind base;
@tailwind components;
@tailwind utilities;

@layer utilities {
    .alert {
        @apply bg-red-50;
        &_title {
            @apply bg-slate-100;
        }
      }
}

but when I run tailwindcss, the nested class was NOT generated properly. 
npx tailwindcss -i ./static/main.css -o output.css

but when I use postcss directly, it worked well.
npx postcss ./static/main.css -o output.css

here is my config files
**tailwind.config.cjs**

  plugins: [
    require('@tailwindcss/aspect-ratio'),
    require('@tailwindcss/typography'),
    require('@tailwindcss/forms'),
    require('@tailwindcss/line-clamp'),
  ],

**postcss.config.cjs**

// postcss.config.js
module.exports = {
  plugins: [
    require('postcss-import'),
    require('@tailwindcss/nesting'),
    require('tailwindcss'),
    require('autoprefixer'),
  ]
}

Thanks in advance

**Comment by tofuwu-dev** [comment]
Hi, You dont need to run many commands and write many codes to use tailwind css...

```bash
# generate postcsss.config.js and tailwind.config.js
npx tailwindcss init -p
```
Plz make sure you installed these dependencies. read this doc

```bash
npm install -D tailwindcss postcss autoprefixer
```

You dont need to reconfigure postcss config file. Leave as is...
```typescript
#postcss.config.js
module.exports = {
  plugins: {
    tailwindcss: {},
    autoprefixer: {},
  },
}
```
Do something necessary here
```typescript
module.exports = {
  content: ["./src/**/*.{js,ts,jsx,tsx}"],
  variants: {},
  extend: {},
  plugins: [
      require('postcss-import'),
      require('@tailwindcss/nesting'),
      require('tailwindcss'),
      require('autoprefixer'),
  ]
```

Tailwind 3.0 supports CDN. You can start tailwind css faster without any installation and command :)
https://tailwindcss.com/blog/tailwindcss-v3#play-cdn

**Reply by kai-wang** [reply]
Thanks for your reply.

CDN doesn't work for me since I am setting up a local dev environment and need to port a list of scss from an old project to this. That is why I need to get nesting working.

I tried to create one new test project with all the configurations you recommended and the tailwindcss give me the same wrong format of output as I mentioned in the original post.

.alert {
  --tw-bg-opacity: 1;
  background-color: rgb(254 242 242 / var(--tw-bg-opacity));
  &_title {
    --tw-bg-opacity: 1;
    background-color: rgb(241 245 249 / var(--tw-bg-opacity))
  }
}

**Reply by tofuwu-dev** [reply]
Can you put your test repo link here?
Why did you use --tw prefix? Have you configured it?

**Reply by kai-wang** [reply]
thanks, --tw is the output css rather than the input css. No prefix is configured in my tailwind.

I committed the repo https://github.com/kai-wang/tailwind_test

The reproduce steps are documented in the README.

Appreciate it!
